stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -173,14 +209,40 @@ private static MethodSpec getWriteMethodSpec(@NotNull TypeName typeName,
                 typeAdapterNamesMap.put(fieldType.toString(), fieldName);
                 String originalFieldName = FileGenUtils.unescapeEscapedString(fieldName);
                 adapterBuilder.addField(typeName, originalFieldName, Modifier.PRIVATE, Modifier.FINAL);
-                constructorBuilder.addStatement(fieldName + " = gson.getAdapter(" +
-                                                typeTokenConstantsGenerator.addTypeToken(fieldType) + ")");
+                constructorBuilder.addStatement(fieldName + " = gson.getAdapter(" + getTypeTokenCode(fieldType, typeVarsMap, typeTokenConstantsGenerator) + ")");
             }
         }
 
         return typeAdapterNamesMap;
     }
 
+    @NotNull
+    private static String getTypeTokenCode(@NotNull TypeMirror fieldType, @NotNull Map<TypeVariable, String> typeVarsMap, @NotNull TypeTokenConstantsGenerator typeTokenConstantsGenerator) {
+        String result = null;
+        if (!TypeUtils.isConcreteType(fieldType)) {
+            if (fieldType.getKind() == TypeKind.TYPEVAR) {
+                result = "(" + ClassName.get(TypeToken.class) + "<" + fieldType + ">)" + ClassName.get(TypeToken.class) + ".get(" + typeVarsMap.get(fieldType) + ")";
+            } else if (fieldType instanceof DeclaredType) {
+                DeclaredType decaredFieldType = (DeclaredType) fieldType;
+                List<? extends TypeMirror> typeMirrors = ((DeclaredType) fieldType).getTypeArguments();
+                result = "com.vimeo.sample.stag.generated." + ParameterizedTypeGenerator.CLASS_NAME + ".getTypeToken(" + decaredFieldType.asElement().toString() + ".class";
```

Comment:
Since the ParameterizedTypeUtils has been moved to the stag-library, this is not required anymore.